datumaro/components/media.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Callable, Optional, Tuple, Union
+import os
+import os.path as osp
+import shutil
+
+from attr import attrs
+import numpy as np
+
+from datumaro.util.image import (
+    _image_loading_errors, decode_image, lazy_image, save_image,
+)
+
+
+@attrs(auto_attribs=True)
+class MediaElement:
+    path: str
+    "Path to the media file"
+
+class Image(MediaElement):
+    def __init__(self,
+            data: Union[np.ndarray, Callable[[str], np.ndarray], None] = None,
+            *,
+            path: Optional[str] = None,
+            size: Optional[Tuple[int, int]] = None):
+        assert size is None or len(size) == 2, size
+        if size is not None:
+            assert len(size) == 2 and 0 < size[0] and 0 < size[1], size
+            size = tuple(map(int, size))
+        self._size = size # (H, W)
+        if not self._size and isinstance(data, np.ndarray):
+            self._size = data.shape[:2]
+
+        assert path is None or isinstance(path, str), path
+        if path is None:
+            path = ''
+        elif path:
+            path = osp.abspath(path).replace('\\', '/')
+        self._path = path
+
+        if not isinstance(data, np.ndarray):
+            assert path or callable(data), "Image can not be empty"
+            assert data is None or callable(data)
+            if path and osp.isfile(path) or data:
+                data = lazy_image(path, loader=data)
+        self._data = data
+
+    @property
+    def path(self) -> str:
+        "Path to the media file"
```

Comment:
Docstrings should always be triple quoted by convention (see [PEP 257](https://www.python.org/dev/peps/pep-0257/)).

Reply:
Fixed